### PR TITLE
Remove `check` from `_prims_common`, replace with `torch._check*`

### PR DESCRIPTION
**Pull request by kurtamohler**
Part of #72948

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225

### CODE REVIEW

**Review by lezcano — APPROVED** [review]
LGTM, but let's wait to see if Alban has anything to say

File: .github/ci_commit_pins/vision.txt [hunk 1]
```diff
@@ -1 +1 @@
-01b9faa16cfeacbb70aa33bd18534de50891786b
+f36c5de4fdaa65b419babcbd4655f4c1a1d31ea6
```
Comment (albanD) [inline .github/ci_commit_pins/vision.txt:1]: Is this done on purpose?

**Review by albanD — APPROVED** [review]
SGTM beyond the submodule update

File: .github/ci_commit_pins/vision.txt (hunk 1 above)
Comment (kurtamohler) [reply .github/ci_commit_pins/vision.txt:1]: Yes, this is the commit hash that my PR in torchvision https://github.com/pytorch/vision/pull/7625 was committed as. It's required to make CI pass

**Comment by izaitsevfb** [comment]
@kurtamohler, we had to revert your PR (see #103128), as it breaks one of internal Meta projects, and, unfortunately, it's impossible to forward fix atm.

I believe @drdarshan and @malfet have more context and understanding on how to proceed with this change.

I apologize for the inconvenience.

**Pull request closed without merging**